app/controllers/tweets_controller.rb

```diff
@@ -0,0 +1,83 @@
+class TweetsController < ApplicationController
+
+	get '/tweets' do
+    if logged_in?
+			@user = current_user
+      @tweets = Tweet.all
+      erb :"tweets/tweets"
+    else
+		  redirect to "/login"
+    end
+	end
+
+	get '/tweets/new' do
+		if !logged_in?
+			redirect to "/login"
+		else
+			erb :"tweets/create_tweet"
+		end
+	end
+
+  post '/tweets' do
+    if logged_in?
+      if params[:content].empty?
+        redirect to "/tweets/new"
+      else
+        # raise params.inspect
+        @tweet = current_user.tweets.create(content: params[:content])
+
+        redirect to "/tweets/#{@tweet.id}"
+      end
+    else
+      redirect to "/login"
+    end
+  end
+
+  get '/tweets/:id' do
+    if logged_in?
+      @tweet = Tweet.find_by(params[:id])
+      erb :"tweets/show_tweet"
+    else
+      redirect to "/login"
+    end
+  end
+
+	get '/tweets/:id/edit' do
+		if !logged_in?
+			redirect to "/login"
+		else
+			@tweet = Tweet.find_by(params[:id])
+			if @tweet && @tweet.user == current_user
+				erb :"tweets/edit_tweet"
+			else
+				redirect to "/tweets"
+			end
+		end
+	end
+
+  patch '/tweets/:id' do
```

Comment:
I would contend that you still want to double check the user's association to the tweet here. There are other ways to submit a PATCH/POST request than going through an HTML form...